### PR TITLE
Fix of logic app standard deployment bug

### LINKED ISSUE #315: Changes not reflected after Logic App Standard deployment
**Describe the bug**
Changes made inside **existing** logic app file (workflow.json, parameters.json, connections.json) deployed via Logic App Standard module (https://github.com/recognizegroup/terraform/tree/develop/modules/azure/logic_app_standard) are not reflected. This causes problem if you want to update/fix logic app from code and deploy the change to an environment.

**To reproduce**
1. in exisiting logic app change parameter.json or workflow.json (add parameter, change action name etc.)
2. deploy this change via Logic App Standard module
3. go to logic app in Azure Portal and check deployment logs
4. OK: deployment has been done
5. go to changed file (paramtere.json or workflow.json) to check the change
6. BUG: nothing has changed

**Expected behavior**
Changes should be reflected after logic app deployment.

**Environment**
any - local PC (Windows 10), Ubuntu (Latest) pipeline

**Notes**
This problem is caused by terraform archive_file module which is changing change timestamp of archived files (https://github.com/hashicorp/terraform-provider-archive/issues/40). Zip file created by the module is deployed via Kudu ZIP deploy (https://learn.microsoft.com/en-us/azure/app-service/deploy-zip?tabs=cli). Kudu is comparing changed timestamps and updates only changed one (see Efficient file copy in https://github.com/projectkudu/kudu/wiki/Deploying-from-a-zip-file-or-url). The issue in archive_file module leads to timestamp is always the same (Jan 1 2049). No changed files are copied by Kudu then. Only new files are copied for the first time.

### PR DESCRIPTION
**Pull request by tjezek**
This is PR to fix logic app standard changes deployment described here https://github.com/recognizegroup/terraform/issues/315. This solution uses local-exec linux shell to crete deploy.zip file with untouched timestamps. The problem is that it's dependent on 'ubuntu-latest' machine which has 'zip' package preinstalled. I don't know how we should solve this OS specific, so I want to discuss this here.

### CODE REVIEW

**Comment by bartwesselink** [comment]
@tjezek nice catch. Is it possible that we just add a check and say that it is currently only supportedf on Ubuntu?

**Review by bartwesselink — APPROVED** [review]

File: modules/azure/logic_app_standard/main.tf [hunk 1]
```diff
@@ -54,11 +58,26 @@ resource "azurerm_logic_app_standard" "app" {
...
+resource "null_resource" "zip_logic_app" {
+  depends_on = [data.archive_file.check_zip]
+
+  triggers = {
+    deploy = data.archive_file.check_zip.output_sha
+  }
+  # if check.zip file changes, create deploy.zip file
+  provisioner "local-exec" {
+    interpreter = local.is_linux ? ["bash", "-c"] : ["PowerShell", "-Command"]
+    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && cd files && cd ${var.workflows_source_path} && zip -rq $OLDPWD/deploy.zip ." : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```
Comment (tom-reinders) [inline modules/azure/logic_app_standard/main.tf:79]: Would not one of these more readable?

```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && zip -rq files/deploy.zip ${var.workflows_source_path}" : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```

```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && zip -rq ${path.module}/files/deploy.zip ${var.workflows_source_path}" : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```

**Review by tom-reinders — CHANGES_REQUESTED** [review]

File: modules/azure/logic_app_standard/main.tf (hunk 1 above)
Comment (tom-reinders) [reply modules/azure/logic_app_standard/main.tf:79]: Would this
```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && cd ${var.workflows_source_path} && zip -rq $OLDPWD/files/deploy.zip ." : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```
or this
```suggestion
    command     = local.is_linux ? "cd ${path.module} && mkdir -p files && cd ${var.workflows_source_path} && zip -rq ${path.module}/files/deploy.zip ." : "New-Item -Path \"${path.module}\" -Name \"files\" -ItemType \"directory\" -Force; Compress-Archive -Path \"${var.workflows_source_path}\\*\" -DestinationPath \"${path.module}\\files\\deploy.zip\""
```
work?

Comment (tjezek) [reply modules/azure/logic_app_standard/main.tf]: The first one works, the second one throws an error.

**Review by tom-reinders — APPROVED** [review]

**Pull request merged**